src/sentry/uptime/grouptype.py

```diff
@@ -1,15 +1,289 @@
+
+    status_reason = result["status_reason"]
+    if status_reason:
+        reason_evidence = IssueEvidence(
+            name="Failure reason",
+            value=f'{status_reason["type"]} - {status_reason["description"]}',
+            important=True,
+        )
+        evidence_display.extend([reason_evidence])
+
+    duration_evidence = IssueEvidence(
+        name="Duration",
+        value=f"{result["duration_ms"]}ms",
+        important=False,
+    )
+    evidence_display.append(duration_evidence)
+
+    request_info = result["request_info"]
+    if request_info:
+        method_evidence = IssueEvidence(
+            name="Method",
+            value=request_info["request_type"],
+            important=False,
+        )
+        status_code_evidence = IssueEvidence(
+            name="Status Code",
+            value=str(request_info["http_status_code"]),
+            important=False,
+        )
+        evidence_display.extend([method_evidence, status_code_evidence])
+
+    return evidence_display
+
+
+def build_event_data(result: CheckResult, detector: Detector) -> EventData:
+    # Default environment when it hasn't been configured
+    env = detector.config.get("environment", "prod")
+
+    # Received time is the actual time the check was performed.
+    received = datetime.fromtimestamp(result["actual_check_time_ms"] / 1000)
+
+    # XXX(epurkhiser): This can be changed over to using the detector ID in the
+    # future once we're no longer using the ProjectUptimeSubscription.id as a tag.
+    project_subscription = get_project_subscription(detector)
+
+    return {
+        "project_id": detector.project_id,
+        "environment": env,
+        "received": received,
+        "platform": "other",
+        "sdk": None,
+        "tags": {
+            "uptime_rule": str(project_subscription.id),
+        },
+        "contexts": {
+            "trace": {"trace_id": result["trace_id"], "span_id": result.get("span_id")},
+        },
+    }
+
+
+class UptimeDetectorHandler(StatefulDetectorHandler[UptimePacketValue, CheckStatus]):
+    @override
+    @property
+    def thresholds(self) -> DetectorThresholds:
+        return {
+            DetectorPriorityLevel.OK: get_active_recovery_threshold(),
+            DetectorPriorityLevel.HIGH: get_active_failure_threshold(),
+        }
+
+    @override
+    def extract_value(self, data_packet: DataPacket[UptimePacketValue]) -> CheckStatus:
+        return data_packet.packet.check_result["status"]
+
+    @override
+    def build_issue_fingerprint(self, group_key: DetectorGroupKey = None) -> list[str]:
+        # TODO(epurkhiser): We should migrate the fingerprints over to match
+        # what the default fingerprint is.
+        return build_fingerprint(self.detector)
+
+    @override
+    def extract_dedupe_value(self, data_packet: DataPacket[UptimePacketValue]) -> int:
+        return int(data_packet.packet.check_result["scheduled_check_time_ms"])
+
+    @override
+    def evaluate(
```

Comment:
High level, I feel we should aim for this evaluate function to be pure. All the side effects and extra conditionals in here are a design smell. I'd encourage us to generalize these:
* metrics should be emitted by framework layer
* feature flags to test things and turn on/off issue creation seem like they're something we'll want for most detectors and the framework should provide this too.

Reply:
Yes I agree. Parts of this have to do with the fact that we’re migrating away from our existing issue creation system for uptime. Once we’ve cleaned up more of that this should simplify and we can clean things up more.